backend/asmgen.ml

```diff
@@ -242,6 +242,22 @@ let test_cfgize (f : Mach.fundecl) (res : Linear.fundecl) : unit =
       f.Mach.fun_name;
   end
 
+let reorder_blocks_random ppf_dump cl =
+  match !Flambda_backend_flags.reorder_blocks_random with
+  | None -> cl
+  | Some seed ->
+     (* Initialize random state based on user-provided seed and function name.
+        Per-function random state (instead of per-compilation unit or
+        per call to ocamlopt) is good for debugging: it gives us
+        deterministic builds for each user-provided seed, regardless
+        of the order of functions in the file or
```

Comment:
Given the discussion, "regardless of the
order of functions in the file" sounds
dubious, maybe it should just be removed.

Reply:
Fixed.  
The alternative of reinitializing random state per compilation unit can be faster. It can be done in `compile_unit` and if needed in `compile_phrase`, which is also accessible by the user.